src/drivers/px4io/px4io_i2c.cpp

```diff
@@ -160,9 +169,23 @@ PX4IO_I2C::read(unsigned address, void *data, unsigned count)
 	msgv[1].buffer = (uint8_t *)values;
 	msgv[1].length = 2 * count;
 
-	int ret = transfer(msgv, 2);
-	if (ret == OK)
-		ret = count;
+	msgv[2].flags = I2C_M_READ | I2C_M_NORESTART;
+	msgv[2].buffer = (uint8_t *)&crc;
+	msgv[2].length = sizeof(crc);
+
+	int ret = transfer(msgv, 3);
+	if (ret == OK) {
+		uint32_t tcrc = 0;
+
+		tcrc = crc32part(&addr[0], 2, tcrc);
```

Comment:
As we we get negative test results, I would assume that things are either off by something here..